### PR TITLE
ssl: detect duplicate client handshake

### PR DESCRIPTION
**Pull request by regit**
Some invalid implementation of TLS have been seen where the client is sending two handshake messages at start. The result was a problem of JA3 generation.

As it is invalid if we follow the RFC, let's ignore the second message.

Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at
   https://docs.suricata.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [x] I have signed the Open Information Security Foundation contribution agreement at
   https://suricata.io/about/contribution-agreement/ (note: this is only required once)
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/6634

Describe changes:
- handle the case where 2 hello message are sent by client

### CODE REVIEW

**Comment by catenacyber** [comment]
Is there a SV test ?

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 17098

File: src/app-layer-ssl.c [hunk 1]
```diff
@@ -1498,6 +1499,16 @@ static int SSLv3ParseHandshakeType(SSLState *ssl_state, const uint8_t *input,
 
     switch (ssl_state->curr_connp->handshake_type) {
         case SSLV3_HS_CLIENT_HELLO:
+            if (ssl_state->flags & SSL_AL_FLAG_STATE_CLIENT_HELLO) {
+                SSLSetEvent(ssl_state, TLS_DECODER_EVENT_DUPLICATE_HANDSHAKE_MESSAGE);
+                if (ssl_state->client_connp.ja3_str) {
```
Comment (victorjulien) [inline src/app-layer-ssl.c:1504]: why is the ja3 stuff freed here?

**Review by victorjulien — CHANGES_REQUESTED** [review]
see inline comments/requests

Please also add the ticket number to the commit.

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 17098

**Comment by victorjulien** [comment]
Is the PCAP in https://redmine.openinfosecfoundation.org/issues/7016 useful for a test @regit ?

**Comment by victorjulien** [comment]
@regit are you planning to revisit this soon?

**Comment by catenacyber** [comment]
My analysis on this :

the PCAP in https://redmine.openinfosecfoundation.org/issues/7016 may be useful because it has indeed 2 client hellos (cf `tls.handshake.type == 1`) but in this case, this is not a faulty SSL implementation : the server sends a Hello Retry Request, and the client sends again its client hello... So, not so sure we should have an event for such legit usages

> As it is invalid if we follow the RFC, let's ignore the second message.

I did not find any mentions of this ja3 RFC for multiple hellos...

The ja3 hash should be safe already thanks to the check `ssl_state->curr_connp->ja3_hash == NULL` : we generate the hash only once after the first TLS hello extensions

**Comment by catenacyber** [comment]
New solution in https://github.com/OISF/suricata/pull/11710

**Pull request closed without merging**